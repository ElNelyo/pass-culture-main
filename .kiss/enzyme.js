// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
