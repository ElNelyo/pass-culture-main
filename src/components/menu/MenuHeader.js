/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'
import PropTypes from 'prop-types'

import { ROOT_PATH } from '../../utils/config'
import { getWalletValue } from '../../utils/user'

const MenuHeader = ({ user }) => {
  const walletValue = getWalletValue(user)
  const avatar = `${ROOT_PATH}/icons/avatar-default-w-XL.svg`
  return (
    <div id="main-menu-header" className="flex-columns is-relative py16 fs18">
      <div className="column-profile text-center">
        <p id="main-menu-header-avatar">
          <img alt="" src={avatar} className="mb3" />
        </p>
        <p
          id="main-menu-header-username"
          className="is-clipped text-ellipsis px5"
        >
          <span>{user && user.publicName}</span>
        </p>
      </div>
      <div className="column-account flex-1 flex-rows flex-center px12">
        <p className="fs30 mb12">
          <span>Mon Pass</span>
        </p>
        <p id="main-menu-header-wallet-value">
          <span className="fs52 is-normal" style={{ lineHeight: '42px' }}>
            {walletValue}&nbsp;â‚¬
          </span>
        </p>
      </div>
    </div>
  )
}

MenuHeader.defaultProps = {
  user: null,
}

MenuHeader.propTypes = {
  user: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
}

export default MenuHeader
