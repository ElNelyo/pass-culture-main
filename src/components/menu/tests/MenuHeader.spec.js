import React from 'react'
import { shallow } from 'enzyme'

import MenuHeader from '../MenuHeader'

describe('src | components | menu | MenuHeader', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = {
        user: true,
      }

      // when
      const wrapper = shallow(<MenuHeader {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
