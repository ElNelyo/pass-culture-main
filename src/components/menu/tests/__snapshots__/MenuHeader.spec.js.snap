// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | menu | MenuHeader snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuHeader
    user={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="column-profile text-center"
        >
          <p
            id="main-menu-header-avatar"
          >
            <img
              alt=""
              className="mb3"
              src="http://localhost/icons/avatar-default-w-XL.svg"
            />
          </p>
          <p
            className="is-clipped text-ellipsis px5"
            id="main-menu-header-username"
          >
            <span />
          </p>
        </div>,
        <div
          className="column-account flex-1 flex-rows flex-center px12"
        >
          <p
            className="fs30 mb12"
          >
            <span>
              Mon Pass
            </span>
          </p>
          <p
            id="main-menu-header-wallet-value"
          >
            <ActivationButton
              className="fs18 py12 is-full-width pc-theme-dark-primary"
              onClickHandler={[Function]}
              style={Object {}}
            />
          </p>
        </div>,
      ],
      "className": "flex-columns is-relative py16 fs18",
      "id": "main-menu-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              id="main-menu-header-avatar"
            >
              <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />
            </p>,
            <p
              className="is-clipped text-ellipsis px5"
              id="main-menu-header-username"
            >
              <span />
            </p>,
          ],
          "className": "column-profile text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />,
              "id": "main-menu-header-avatar",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "mb3",
                "src": "http://localhost/icons/avatar-default-w-XL.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span />,
              "className": "is-clipped text-ellipsis px5",
              "id": "main-menu-header-username",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="fs30 mb12"
            >
              <span>
                Mon Pass
              </span>
            </p>,
            <p
              id="main-menu-header-wallet-value"
            >
              <ActivationButton
                className="fs18 py12 is-full-width pc-theme-dark-primary"
                onClickHandler={[Function]}
                style={Object {}}
              />
            </p>,
          ],
          "className": "column-account flex-1 flex-rows flex-center px12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Mon Pass
              </span>,
              "className": "fs30 mb12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Mon Pass",
              },
              "ref": null,
              "rendered": "Mon Pass",
              "type": "span",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                false,
                <ActivationButton
                  className="fs18 py12 is-full-width pc-theme-dark-primary"
                  onClickHandler={[Function]}
                  style={Object {}}
                />,
              ],
              "id": "main-menu-header-wallet-value",
            },
            "ref": null,
            "rendered": Array [
              false,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": null,
                  "className": "fs18 py12 is-full-width pc-theme-dark-primary",
                  "onClickHandler": [Function],
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="column-profile text-center"
          >
            <p
              id="main-menu-header-avatar"
            >
              <img
                alt=""
                className="mb3"
                src="http://localhost/icons/avatar-default-w-XL.svg"
              />
            </p>
            <p
              className="is-clipped text-ellipsis px5"
              id="main-menu-header-username"
            >
              <span />
            </p>
          </div>,
          <div
            className="column-account flex-1 flex-rows flex-center px12"
          >
            <p
              className="fs30 mb12"
            >
              <span>
                Mon Pass
              </span>
            </p>
            <p
              id="main-menu-header-wallet-value"
            >
              <ActivationButton
                className="fs18 py12 is-full-width pc-theme-dark-primary"
                onClickHandler={[Function]}
                style={Object {}}
              />
            </p>
          </div>,
        ],
        "className": "flex-columns is-relative py16 fs18",
        "id": "main-menu-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                id="main-menu-header-avatar"
              >
                <img
                  alt=""
                  className="mb3"
                  src="http://localhost/icons/avatar-default-w-XL.svg"
                />
              </p>,
              <p
                className="is-clipped text-ellipsis px5"
                id="main-menu-header-username"
              >
                <span />
              </p>,
            ],
            "className": "column-profile text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt=""
                  className="mb3"
                  src="http://localhost/icons/avatar-default-w-XL.svg"
                />,
                "id": "main-menu-header-avatar",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "mb3",
                  "src": "http://localhost/icons/avatar-default-w-XL.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
                "className": "is-clipped text-ellipsis px5",
                "id": "main-menu-header-username",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="fs30 mb12"
              >
                <span>
                  Mon Pass
                </span>
              </p>,
              <p
                id="main-menu-header-wallet-value"
              >
                <ActivationButton
                  className="fs18 py12 is-full-width pc-theme-dark-primary"
                  onClickHandler={[Function]}
                  style={Object {}}
                />
              </p>,
            ],
            "className": "column-account flex-1 flex-rows flex-center px12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Mon Pass
                </span>,
                "className": "fs30 mb12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mon Pass",
                },
                "ref": null,
                "rendered": "Mon Pass",
                "type": "span",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  false,
                  <ActivationButton
                    className="fs18 py12 is-full-width pc-theme-dark-primary"
                    onClickHandler={[Function]}
                    style={Object {}}
                  />,
                ],
                "id": "main-menu-header-wallet-value",
              },
              "ref": null,
              "rendered": Array [
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": null,
                    "className": "fs18 py12 is-full-width pc-theme-dark-primary",
                    "onClickHandler": [Function],
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
